doc/nat/nat_design_spec.md

```diff
@@ -0,0 +1,1422 @@
+    "entry_type"            : {{static_or_dynamic}}
+     
+NAT_GLOBAL_TABLE:Values
+    "admin_mode"     : {{enable_or_disable}}
+    "nat_timeout"    : {{timeout_in_secs}}
+    "nat_tcp_timeout": {{timeout_in_secs}}
+    "nat_udp_timeout": {{timeout_in_secs}}
+```
+
+### 3.2.4 APP DB Schemas
+
+```
+; Defines schema for the NAPT translation entries
+key                                   = NAPT_TABLE:ip_protocol:ip:l4_port     ; NAPT table
+; field                               = value
+TRANSLATED_IP                         = ipv4
+TRANSLATED_L4_PORT                    = port_num
+NAT_TYPE                              = "snat" / "dnat"
+ENTRY_TYPE                            = "static" / "dynamic"
+
+; value annotations
+port_num      = 1*5DIGIT   ; a number between 1 and 65535
+```
+
+```
+; Defines schema for the NAT translation entries
+key                                   = NAT_TABLE:ip           ; NAT table
+; field                               = value
+TRANSLATED_IP                         = ipv4
+NAT_TYPE                              = "snat" / "dnat"
+ENTRY_TYPE                            = "static" / "dynamic"
+```
+
+```
+; Defines schema for the Twice NAT translation entries
+
+key                                   = NAT_TWICE_TABLE:src_ip:dst_ip
+TRANSLATED_SRC_IP                     = ipv4
+TRANSLATED_DST_IP                     = ipv4
+ENTRY_TYPE                            = "static" / "dynamic" 
+```
+
+```
+; Defines schema for the Twice NAPT translation entries
+
+key                                   = NAPT_TWICE_TABLE:ip_protocol:src_ip:src_l4_port:dst_ip:dst_l4_port
+TRANSLATED_SRC_IP                     = ipv4
+TRANSLATED_SRC_L4_PORT                = port_num
+TRANSLATED_DST_IP                     = ipv4
+TRANSLATED_DST_L4_PORT                = port_num
+ENTRY_TYPE                            = "static" / "dynamic" 
+```
+
+```
+; Defines schema for the NAT global table
+key                                   = NAT_GLOBAL_TABLE:Values        ; NAT global table
+; field                               = value
+ADMIN_MODE                     = "enable" / "disable" ; If the NAT feature is enabled or disabled globally
+NAT_TIMEOUT                    = 1*6DIGIT         ; Timeout in secs (Range: 300 sec - 432000 sec)
+NAT_TCP_TIMEOUT                = 1*3DIGIT         ; Timeout in secs (Range: 120 sec - 600 sec)
+NAT_UDP_TIMEOUT                = 1*6DIGIT         ; Timeout in secs (Range: 300 sec - 432000 sec)
+```
+
+### 3.2.5 COUNTERS DB
+The following new counters are available per zone:
+```
+COUNTERS:NAT_ZONE:<zone-id>
+    SAI_NAT_DNAT_DISCARDS
+    SAI_NAT_SNAT_DISCARDS
+    SAI_NAT_DNAT_TRANSLATION_NEEDED
+    SAI_NAT_SNAT_TRANSLATION_NEEDED
+    SAI_NAT_DNAT_TRANSLATIONS
+    SAI_NAT_SNAT_TRANSLATIONS
+
+```
+The following new counters are available per entry:
+```
+The counters in the COUNTERS_DB are updated every 10 seconds. The key for the entry in the COUNTERS_DB is same as the key in the APP_DB.
```

Comment:
potentially, there are lots counters to pull, can you look into the flex counter design to see how can we reduce the counter pulling overhead?

Reply:
Sure, will have a look.

Reply:
Will move to flex counters once we scale up the NAT entries beyond 1K in the near future.